API-227766 Replace deprecated integration API calls with new Modules API

ov-automation opened the pull request:
[API-227766]( https://trackor.onevizion.com/form/ConfigAppForm.do?id=100112400743&ttid=10009161) Replace deprecated integration API calls with new Modules API<br><br>[API-227766-147245](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100112740684&ttid=100091445) vadim.glebov (Vadim Glebov)<br>[API-227766-147247](https://trackor.onevizion.com/form/ConfigAppForm.do?id=100112740685&ttid=100091445) amoiseenko (Andrey Moiseenko)

Anastasiyaaa committed:
API-227766 Replace deprecated integration API calls with new Modules API
API-227766-144450 [DEV] Replace deprecated integration API calls with new Modules API

Anastasiyaaa committed:
API-227766 Replace deprecated integration API calls with new Modules API
API-227766-144450 [DEV] Replace deprecated integration API calls with new Modules API

glebowadim commented on onevizion/integration/log.py:
```diff
@@ -1,55 +0,0 @@
-from onevizion.httpbearer import HTTPBearerAuth
-import onevizion
-
-class IntegrationLog(object):
```
I think we should not remove old classes right now and we should mark them as deprecated. But old classes should also use new API endpoints.

Anastasiyaaa replied:
I did not delete the old classes, but renamed the **integration** directory to **module**, because it won't affect anything
Please pay attention to the **onevizion/module/log.py** file, there is still an **IntegrationLog** class there

glebowadim replied:
I see

glebowadim commented on onevizion/module/log.py:
```diff
@@ -0,0 +1,112 @@
+
+	def __init__(self, processId, URL="", userName="", password="", paramToken=None, isTokenAuth=False, logLevelName="Error"):
+		"""This throws a deprecation warning on initialization."""
+		warn(f'{self.__class__.__name__} will be deprecated.', DeprecationWarning, stacklevel=2)
```
' is deprecated'

Anastasiyaaa replied:
Fixed

glebowadim commented on onevizion/module/log.py:
```diff
@@ -0,0 +1,112 @@
+
+# deprecated has been used since version 3.13
+# @deprecated("Use ModuleLog instead")
+class IntegrationLog(object):
```
Looks like both of these classes have the same code. 
Propose to create an instance of ModuleLog in IntegrationLog and use it.

Anastasiyaaa replied:
Done

Anastasiyaaa committed:
API-227766 Replace deprecated integration API calls with new Modules API
API-227766-147245 [CR] Replace deprecated integration API calls with new Modules API

Anastasiyaaa commented:
Please note that `onevizion\notif\service.py` uses `IntegrationLog`, I did not change this in the current Issue

glebowadim commented:
> Please note that `onevizion\notif\service.py` uses `IntegrationLog`, I did not change this in the current Issue

Why?

Anastasiyaaa commented:
> > Please note that `onevizion\notif\service.py` uses `IntegrationLog`, I did not change this in the current Issue
> 
> Why?

It seemed to me that this does not really apply to the current Issue. In addition, the `NotificationService` class has a property `self._integrationLog`, which cannot be renamed now, because it is used at least in the **sms-notification** module.
Should I replace the use of `IntegrationLog` with `ModuleLog` (in onevizion\notif\service.py) in the current Issue?

glebowadim commented:
> > > Please note that `onevizion\notif\service.py` uses `IntegrationLog`, I did not change this in the current Issue
> > 
> > 
> > Why?
> 
> It seemed to me that this does not really apply to the current Issue. In addition, the `NotificationService` class has a property `self._integrationLog`, which cannot be renamed now, because it is used at least in the **sms-notification** module. Should I replace the use of `IntegrationLog` with `ModuleLog` (in onevizion\notif\service.py) in the current Issue?

Ok let's create 2 Issues for API-v3 and sms-notification and merge them at the same time

glebowadim commented:
> > > Please note that `onevizion\notif\service.py` uses `IntegrationLog`, I did not change this in the current Issue
> > 
> > 
> > Why?
> 
> It seemed to me that this does not really apply to the current Issue. In addition, the `NotificationService` class has a property `self._integrationLog`, which cannot be renamed now, because it is used at least in the **sms-notification** module. Should I replace the use of `IntegrationLog` with `ModuleLog` (in onevizion\notif\service.py) in the current Issue?

I think we can create field of `ModuleLog` in `NotificationService` after that change sms-notification Module. When nobody uses `IntegrationLog` in sms-notification, we will remove it. 

Yes, I think we need to do it in this Issue

Anastasiyaaa committed:
API-227766 Replace deprecated integration API calls with new Modules API
API-227766-147245 [CR] Replace deprecated integration API calls with new Modules API

Anastasiyaaa committed:
API-227766 Replace deprecated integration API calls with new Modules API
API-227766-147245 [CR] Replace deprecated integration API calls with new Modules API

Anastasiyaaa commented:
> > > > Please note that `onevizion\notif\service.py` uses `IntegrationLog`, I did not change this in the current Issue
> > > 
> > > 
> > > Why?
> > 
> > 
> > It seemed to me that this does not really apply to the current Issue. In addition, the `NotificationService` class has a property `self._integrationLog`, which cannot be renamed now, because it is used at least in the **sms-notification** module. Should I replace the use of `IntegrationLog` with `ModuleLog` (in onevizion\notif\service.py) in the current Issue?
> 
> I think we can create field of `ModuleLog` in `NotificationService` after that change sms-notification Module. When nobody uses `IntegrationLog` in sms-notification, we will remove it.
> 
> Yes, I think we need to do it in this Issue

Done
